docs/source/guides/quickstart.md

```diff
@@ -1,48 +1,149 @@
+## Connecting to the NuCypher Network
+
+### Running an Ethereum Node
+
+Operation of nucypher cryptological characters `Alice`, `Bob`, and `Ursula` require
+a running ethereum node and wallet (We recommend Geth).
+
+To run a Goerli connected Geth node:
 ```bash
-$ pip3 install -U nucypher
+$ geth --goerl
 ```
 
-### Run a Federated-Only Development Ursula
+### Provider URI
 
-```bash
-$ nucypher ursula run --dev --federated-only
+Nucypher uses the node's IPC-File to communicate, specified by `provider_uri`.
+By default in ubuntu, the path is `~/.ethereum/goerli/geth.ipc` - This path
+will also be logged to the geth-running console on startup. 
+
+### Ursula: Untrusted Re-Encryption Proxies
+
+When initializing an `Alice`, `Bob`, or `Ursula`, an initial "Stranger-`Ursula`" is needed to perform 
+the role of a `Teacher`, or "seednode".  The `known_nodes` will inform your character of all of the nodes
+they know about network-wide, then kick-off the automated node-discovery loop.
+
+Stranger `Ursula`s can be created by invoking the `from_seed_and_stake_info` method, then a `list` of `known_nodes`
+can be passed into any `Character`'s init:
+```python
+from nucypher.characters.lawful import Ursula, Alice
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri=<URI>)  # ie. https://0.0.0.0:9151
+another_ursula = Ursula.from_seed_and_stake_info(seed_uri=<URI>)
+
+alice = Alice(checksum_address='0xdeadbeef',
+              provider_uri=<PROVIDER URI>,
+              known_nodes=[ursula, another_ursula])
 ```
 
-### Configure a Persistent Ursula
+## Alice: Grant Access to a Secret
 
-```bash
-$ nucypher ursula init --federated-only
+```python
+from nucypher.characters.lawful import Alice, Bob, Ursula
+
+# Application Side-Channel
+# --------------------------
+# bob_encrypting_key = <Side Channel>
+# bob_verifying_key = <Side Channel>
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri='https://0.0.0.0:9151')
+alice = Alice(checksum_address='0xdeadbeef', 
+              provider_uri=<PROVIDER URI>,
+              known_nodes=[ursula])
+bob = Bob.from_public_keys(verifying_key=bob_verifying_key, encrypting_key=bob_encrypting_key)
 ```
 
-### Run a Persistent Ursula
+```python
+from datetime import datetime, timedelta
 
-```bash
-$ nucypher ursula run --network <NETWORK_DOMAIN> --teacher <SEEDNODE_URI> --federated-only
+#
+# Grant
+#
+
+expiration = datetime.now() + timedelta(days=5)  # Five days from now.
+policy = alice.grant(bob,
+                     label=b'my-secret-stuff',
+                     m=2, n=3,
+                     expiration=expiration)
+
+policy_encrypting_key = policy.public_key
 ```
 
-Replace `<NETWORK_DOMAIN>` with the network domain and `<SEEDNODE_URI>` with the URI of a node running on that network
-domain you want to connect to (for example `0.0.0.0:9151` or `0xdeadbeef@0.0.0.0:9151`).
+Note that Alice can get the public key even before creating the policy.
+From this moment on, any Data Source (Enrico) that knows the public key
```

Comment:
I'd move this introduction into the Enrico section below

Reply:
:heavy_check_mark: